shared/react-components/components/Modals/DelegateModal/DelegateTxButton.tsx

```diff
@@ -0,0 +1,175 @@
+  }
+}
+
+export const DelegateTxButton = (props: DelegateTxButtonProps) => {
+  const {
+    twabController,
+    vault,
+    modalView,
+    setModalView,
+    openConnectModal,
+    openChainModal,
+    addRecentTransaction,
+    onSuccessfulDelegation,
+    intl
+  } = props
+
+  const { address: userAddress, isDisconnected } = useAccount()
+  const { chain } = useNetwork()
+
+  const { data: tokenData } = useVaultTokenData(vault)
+
+  const newDelegateAddress: Address | undefined = useAtomValue(delegateFormNewDelegateAddressAtom)
+
+  const { data: delegate, refetch: refetchUserVaultDelegate } = useUserVaultDelegate(
+    vault,
+    userAddress as Address,
+    { refetchOnWindowFocus: true }
+  )
+
+  const createToast = (delegateTxHash: Address) => {
+    if (!!vault && !!delegateTxHash) {
+      createDelegateTxToast({
+        vault: vault,
+        txHash: delegateTxHash,
+        addRecentTransaction: addRecentTransaction,
+        intl: intl?.txToast
+      })
+    }
+  }
+
+  const {
+    isWaiting: isWaitingDelegation,
+    isConfirming: isConfirmingDelegation,
+    isSuccess: isSuccessfulDelegation,
+    txHash: delegateTxHash,
+    sendDelegateTransaction
+  } = useSendDelegateTransaction(twabController, newDelegateAddress, vault, {
+    onSend: () => {
+      setModalView('confirming')
+    },
+    onSuccess: () => {
+      refetchUserVaultDelegate()
+      onSuccessfulDelegation?.()
+      setModalView('main')
+    },
+    onError: () => {
+      setModalView('main')
+    }
+  })
+
+  useEffect(() => {
+    if (isWaitingDelegation && modalView !== 'waiting') {
+      setModalView('waiting')
+    }
+  }, [isWaitingDelegation])
+
+  useEffect(() => {
+    if (
+      !!delegateTxHash &&
+      isConfirmingDelegation &&
+      !isWaitingDelegation &&
+      !isSuccessfulDelegation
+    ) {
+      if (delegateTxHash) {
+        createToast(delegateTxHash)
+      }
+      setModalView('confirming')
+    }
+  }, [delegateTxHash, isConfirmingDelegation])
+
+  const delegateAddressHasChanged = newDelegateAddress !== delegate
+
+  const delegateEnabled =
+    !isDisconnected &&
+    !!userAddress &&
+    !!newDelegateAddress &&
+    isAddress(newDelegateAddress) &&
+    !isWaitingDelegation &&
+    !isConfirmingDelegation &&
+    delegateAddressHasChanged &&
+    !!sendDelegateTransaction
+
+  if (isDisconnected) {
+    return null
+  } else if (isWaitingDelegation) {
+    return (
+      <Button color='transparent' fullSized={true} disabled={true}>
+        <Spinner />{' '}
+        <span className='inline-block ml-2'>
+          {intl?.base?.('confirmNotice') ?? 'Confirm Transaction in Wallet'}
+        </span>
+      </Button>
+    )
+  } else if (isConfirmingDelegation) {
+    return (
+      <Button color='transparent' fullSized={true} disabled={true}>
+        <Spinner />
+        <span className='inline-block ml-2'>
+          {intl?.base?.('waiting') ?? 'Waiting for Transaction'}
+        </span>
+      </Button>
+    )
```

Comment:
Are these cases necessary or not already handled by "TransactionButton"?

Reply:
Since we are primarily using the button (and toasts) to update the user on state, Alex thought it best to have more descriptive text on that button - and I'm following his designs. I think it makes sense to have these extra states as the TransactionButton component does show a spinner, but doesn't say 'Waiting for Transaction', or 'Confirm Transaction in Wallet'. Perhaps we could add this copy to the TransactionButton component.